src/main/java/com/nike/cerberus/security/VaultAuthPrincipal.java

```diff
@@ -81,7 +77,7 @@ public VaultAuthPrincipal(VaultClientTokenResponse clientToken) {
 
     private Set<String> extractUserGroups(final VaultClientTokenResponse clientToken) {
         final Map<String, String> meta = clientToken.getMeta();
-        final String groupString = meta.get(METADATA_KEY_GROUPS);
+        final String groupString = meta == null ? "" : meta.get(METADATA_KEY_GROUPS);
```

Comment:
The object can be null, so if it is we want it to be empty, or we want to get the groups from the metadata.